RecoEgamma/EgammaTools/src/SCEnergyCorrectorSemiParm.cc

```diff
@@ -294,9 +297,22 @@ void SCEnergyCorrectorSemiParm::modifyObject(reco::SuperCluster &sc) {
     double ecor = mean*eval[6];
     if (!iseb)  
       ecor = mean*eval[6]+sc.preshowerEnergy();
+
+	p.first  = ecor;
+	p.second = 0.;
+
+  }
+
+  return p;
+}
    
-    sc.setEnergy(ecor);
-    sc.setCorrectedEnergy(ecor);
-  }  
+//--------------------------------------------------------------------------------------------------
+void SCEnergyCorrectorSemiParm::modifyObject(reco::SuperCluster &sc) {
+  
+	std::pair<double, double> cor = getCorrections(sc);
+	if(cor.first<0 || cor.second < 0) return;
+	sc.setEnergy(cor.first);
+	sc.setCorrectedEnergy(cor.first);
+	sc.setCorrectedEnergyUncertainty(cor.second);
```

Comment:
It's true. In my understanding the energy uncertainty is not used at HLT. If it is not set by this class, then it should have a default value.
I can add an if(! isHLT) at L316 to preserve the value.